qa/integration-tests/src/test/java/io/camunda/it/utils/CamundaExternalDBExtension.java

```diff
@@ -0,0 +1,152 @@
+  public void beforeAll(final ExtensionContext context) {
+    LOGGER.info("Starting up Camunda instance, with {}", databaseType);
+    final Class<?> testClass = context.getRequiredTestClass();
+    testPrefix = testClass.getSimpleName().toLowerCase();
+
+    switch (databaseType) {
+      case ES -> testApplication.withElasticsearchSupport(DEFAULT_ES_URL, testPrefix);
+      case OS ->
+          testApplication.withOpensearchSupport(
+              DEFAULT_OS_URL, testPrefix, DEFAULT_OS_ADMIN_USER, DEFAULT_OS_ADMIN_PW);
+      case RDBMS -> testApplication.withRdbmsExporter();
+      default -> throw new RuntimeException("Unknown exporter type");
+    }
+    testApplication.start();
+    testApplication.awaitCompleteTopology();
+
+    injectFields(testClass, null, ModifierSupport::isStatic);
+  }
+
+  private void injectFields(
+      final Class<?> testClass, final Object testInstance, Predicate<Field> predicate) {
+    predicate = predicate.and(field -> field.getType() == CamundaClient.class);
+    for (final Field field : testClass.getDeclaredFields()) {
+      try {
+        if (predicate.test(field)) {
+          field.setAccessible(true);
+          field.set(testInstance, createCamundaClient());
+        }
+      } catch (final Exception ex) {
+        throw new RuntimeException(ex);
+      }
+    }
+  }
+
+  @Override
+  public void afterAll(final ExtensionContext context) {
+    if (databaseType == DatabaseType.ES || databaseType == DatabaseType.OS) {
+      final URI deleteEndpoint = URI.create(String.format("%s/%s*", DEFAULT_ES_URL, testPrefix));
+      final HttpRequest httpRequest = HttpRequest.newBuilder().DELETE().uri(deleteEndpoint).build();
+      try (final HttpClient httpClient = HttpClient.newHttpClient()) {
+        final HttpResponse<String> response = httpClient.send(httpRequest, BodyHandlers.ofString());
+        final int statusCode = response.statusCode();
+        if (statusCode / 100 == 2) {
+          LOGGER.info("Test data deleted.");
+        } else {
+          LOGGER.warn("Failure on deleting test data. Status code: {}", statusCode);
+        }
+      } catch (final IOException | InterruptedException e) {
+        LOGGER.warn("Failure on deleting test data.", e);
+      }
+    }
+
+    closeables.parallelStream()
+        .forEach(
+            c -> {
+              try {
+                c.close();
+              } catch (final Exception e) {
+                throw new RuntimeException(e);
+              }
+            });
+  }
+
+  @Override
+  public boolean supportsParameter(
+      final ParameterContext parameterContext, final ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    return parameterContext.getParameter().getType() == CamundaClient.class;
+  }
+
+  @Override
+  public Object resolveParameter(
+      final ParameterContext parameterContext, final ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    return createCamundaClient();
```

Comment:
:question: If the client is injected, who is closing it? The extension here doesn't keep track of it, so the test method has to close it? Might be safer to keep track of it and just close it ourselves on an `afterEach` :shrug: